backend/dataall/tasks/data_sharing/share_processors/s3_process_share.py

```diff
@@ -0,0 +1,186 @@
+        cls,
+        session,
+        dataset: models.Dataset,
+        share: models.ShareObject,
+        share_folders: [models.DatasetStorageLocation],
+        source_environment: models.Environment,
+        target_environment: models.Environment,
+        source_env_group: models.EnvironmentGroup,
+        env_group: models.EnvironmentGroup
+    ) -> bool:
+        """
+        1) (one time only) manage_bucket_policy - grants permission in the bucket policy
+        2) grant_target_role_access_policy
+        3) manage_access_point_and_policy
+        4) update_dataset_bucket_key_policy
+        5) update_share_item_status
+
+        Returns
+        -------
+        True if share is approved successfully
+        """
+        log.info(
+            '##### Starting Sharing folders #######'
+        )
+        for folder in share_folders:
+            log.info(f'sharing folder: {folder}')
+            sharing_item = api.ShareObject.find_share_item_by_folder(
+                session,
+                share,
+                folder,
+            )
+            shared_item_SM = api.ShareItemSM(models.ShareItemStatus.Share_Approved.value)
+            new_state = shared_item_SM.run_transition(models.Enums.ShareObjectActions.Start.value)
+            shared_item_SM.update_state_single_item(session, sharing_item, new_state)
+
+            sharing_folder = cls(
+                session,
+                dataset,
+                share,
+                folder,
+                source_environment,
+                target_environment,
+                source_env_group,
+                env_group,
+            )
+
+            try:
+                sharing_folder.manage_bucket_policy()
+                sharing_folder.grant_target_role_access_policy()
+                sharing_folder.manage_access_point_and_policy()
+                sharing_folder.update_dataset_bucket_key_policy()
+
+                new_state = shared_item_SM.run_transition(models.Enums.ShareItemActions.Success.value)
+                shared_item_SM.update_state_single_item(session, sharing_item, new_state)
+
+            except Exception as e:
+                sharing_folder.handle_share_failure(e)
+
+                new_state = shared_item_SM.run_transition(models.Enums.ShareItemActions.Failure.value)
+                shared_item_SM.update_state_single_item(session, sharing_item, new_state)
+
+        return True
+
+    @classmethod
+    def process_revoked_shares(
+            cls,
+            session,
+            dataset: models.Dataset,
+            share: models.ShareObject,
+            revoke_folders: [models.DatasetStorageLocation],
+            source_environment: models.Environment,
+            target_environment: models.Environment,
+            source_env_group: models.EnvironmentGroup,
+            env_group: models.EnvironmentGroup
+    ) -> bool:
+        """
+        1) (one time only) manage_bucket_policy - grants permission in the bucket policy
```

Comment:
The docs here for the process_revoked_shares do not match, I would suggest the following:
`        """
        1) Remove revoked folder from S3 access point policy
        2) Update_share_item_status

        Returns
        -------
        True if all revoke_folders were revoked successfully
        """`

Reply:
On it :)